### PR TITLE
Scenes as assets

### PR DESCRIPTION
**Pull request by tcdrop**
#863fw7zzt

Scenes are displayed in the asset browser. They can be dragged and dropped or added to an existing entity via a scene component.

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-863fw7zzt Display scenes in asset browser](https://app.clickup.com/t/863fw7zzt)

File: Assets/Scripts/Command/AddAssetToScene.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            this.entityId = entityId;
+            this.entityCustomName = entityCustomName;
+            this.assetId = assetId;
+            this.positionInScene = positionInScene;
+        }
+
+        public override void Undo(DclScene sceneState, EditorEvents editorEvents)
+        {
+            EntityUtility.DeleteEntity(sceneState, entityId);
+            editorEvents.InvokeHierarchyChangedEvent();
```
Comment (cblech) [inline Assets/Scripts/Command/AddAssetToScene.cs:31]: When undoing, you also need to undo the selection state

File: Assets/Scripts/SceneState/DclSceneComponent.cs [hunk 2]
```diff
@@ -8,12 +8,12 @@ public class DclSceneComponent : DclComponent
             new ComponentDefinition(
                 "Scene",
                 "Scene",
-                new DclComponentProperty.PropertyDefinition("sceneId", DclComponentProperty.PropertyType.String, ""));
+                new DclComponentProperty.PropertyDefinition("scene", DclComponentProperty.PropertyType.Asset, Guid.Empty));
```
Comment (cblech) [inline Assets/Scripts/SceneState/DclSceneComponent.cs:11]: This makes previews saves incompatible. Please convert scenes before loading

**Review by cblech — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Command/AddAssetToScene.cs (hunk 1 above)
Comment (tcdrop) [reply Assets/Scripts/Command/AddAssetToScene.cs:31]: The previous selection is now properly saved and gets restored when the command is undone.

File: Assets/Scripts/SceneState/DclSceneComponent.cs (hunk 2 above)
Comment (tcdrop) [reply Assets/Scripts/SceneState/DclSceneComponent.cs:11]: Whenever the verification of the component definition fails, it now tries to use the old definition. If the old one gets verified successfully, the scene component is converted into the new format.

**Review by cblech — APPROVED** [review]

**Comment by MGH-Automator** [comment]
Task linked: [CU-861md77x9 Write a system to upgrade files](https://app.clickup.com/t/861md77x9)

**Comment by MGH-Automator** [comment]
Task linked: [CU-863g2p6qe Undo - Deleting an object doesn't remove its gizmo](https://app.clickup.com/t/863g2p6qe)

**Pull request merged**